test-requirements.txt

```diff
@@ -6,6 +6,7 @@ distro; python_version >= '3.8'
 nose; python_version <= '3.9'
 nose-timer; python_version >= '2.7' and python_version <= '3.9'
 pytest; python_version >= '3.10'
+setuptools; python_version >= '3.12'
```

Comment:
Why setuptools require only on 3.12+?

Reply:
it needs to be installed explicitly on 3.12m it's available by default on other versions.